src/CharsetDetector.cs

```diff
@@ -241,6 +284,37 @@ private static void ReadStream(Stream stream, long? maxBytes, CharsetDetector de
             }
         }
 
+        private static async Task ReadStreamAsync(Stream stream, long? maxBytes, CharsetDetector detector, CancellationToken cancellationToken = default)
+        {
+            const int bufferSize = 1024;
+            byte[] buff = new byte[bufferSize];
+            int read;
+            long readTotal = 0;
+
+            var toRead = CalcToRead(maxBytes, readTotal, bufferSize);
+
+            while ((read = await stream.ReadAsync(buff, 0, toRead, cancellationToken)) > 0)
+            {
+                detector.Feed(buff, 0, read);

```

Comment:
Please deduplicate the code of the `while` loop (exact same in other method?)

Reply:
Resolved and arranged.

Reply:
Deduplicated.